extensions/wikia/Captcha/Captcha.class.php

```diff
@@ -0,0 +1,622 @@
+		$wg = \F::app()->wg;
+		$source = wfMessage( 'captcha-addurl-whitelist' )->inContentLanguage()->text();
+
+		$whitelist = wfEmptyMsg( 'captcha-addurl-whitelist', $source )
+			? false
+			: $this->buildRegexes( explode( "\n", $source ) );
+
+		$cwl = $wg->CaptchaWhitelist !== false ? preg_match( $wg->CaptchaWhitelist, $url ) : false;
+		$wl = $whitelist !== false ? preg_match( $whitelist, $url ) : false;
+
+		return !( $cwl || $wl );
+	}
+
+	/**
+	 * Build regex from white-list
+	 *
+	 * @param string $lines Lines from [[MediaWiki:Captcha-addurl-whitelist]]
+	 *
+	 * @return string Regex or bool false if white-list is empty
+	 */
+	private function buildRegexes( $lines ) {
+		# Code duplicated from the SpamBlacklist extension (r19197)
+
+		# Strip comments and whitespace, then remove blanks
+		$lines = array_filter( array_map( 'trim', preg_replace( '/#.*$/', '', $lines ) ) );
+
+		# No lines, don't make a regex which will match everything
+		if ( count( $lines ) == 0 ) {
+			$this->log( "No lines\n" );
+			return false;
+		} else {
+			# Make regex
+			# It's faster using the S modifier even though it will usually only be run once
+			// $regex = 'http://+[a-z0-9_\-.]*(' . implode( '|', $lines ) . ')';
+			// return '/' . str_replace( '/', '\/', preg_replace('|\\\*/|', '/', $regex) ) . '/Si';
+			$regexes = '';
+			$regexStart = '/^https?:\/\/+[a-z0-9_\-.]*(';
+			$regexEnd = ')/Si';
+			$regexMax = 4096;
+			$build = false;
+			foreach ( $lines as $line ) {
+				// FIXME: not very robust size check, but should work. :)
+				if ( $build === false ) {
+					$build = $line;
+				} elseif ( strlen( $build ) + strlen( $line ) > $regexMax ) {
+					$regexes .= $regexStart .
+						str_replace( '/', '\/', preg_replace( '|\\\*/|', '/', $build ) ) .
+						$regexEnd;
+					$build = $line;
+				} else {
+					$build .= '|' . $line;
+				}
+			}
+			if ( $build !== false ) {
+				$regexes .= $regexStart .
+					str_replace( '/', '\/', preg_replace( '|\\\*/|', '/', $build ) ) .
+					$regexEnd;
+			}
+			return $regexes;
+		}
+	}
+
+	/**
+	 * Load external links from the externallinks table
+	 *
+	 * @param \Title $title
+	 *
+	 * @return Array
+	 */
+	public function getLinksFromTracker( $title ) {
+		$dbr = wfGetDB( DB_SLAVE );
+		$id = $title->getArticleID(); // should be zero queries
+		$res = $dbr->select(
+			'externallinks',
+			[ 'el_to' ],
+			[ 'el_from' => $id ],
+			__METHOD__
+		);
+		$links = [];
+		foreach ( $res as $row ) {
+			/** @var \stdClass $row */
+			$links[] = $row->el_to;
+		}
+		return $links;
+	}
+
+	/**
+	 * Backend function for confirmEdit() and confirmEditAPI()
+	 *
+	 * @param \EditPage $editPage
+	 * @param string $newText
+	 * @param string $section
+	 * @param bool $merged
+	 *
+	 * @return bool false if the CAPTCHA is rejected, true otherwise
+	 */
+	private function doConfirmEdit( $editPage, $newText, $section, $merged = false ) {
+		$wg = \F::app()->wg;
+		if ( $wg->Request->getVal( 'captchaid' ) ) {
```

Comment:
They are the same.  I like to keep the casing of globals the same when I break them apart like this, e.g.:
`$wgRequest`
becomes
`$wg->Request`
Its easier to write and you don't have to change capitalization to grep for either version.